nvflare/edge/assessors/model_manager.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Dict, Optional
+
+import numpy as np
+
+from nvflare.apis.dxo import DXO, DataKind
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.abstract.model import make_model_learnable
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.app_event_type import AppEventType
+from nvflare.edge.aggregators.model_update_dxo import ModelUpdateDXOAggregator
+from nvflare.edge.mud import ModelUpdate
+
+
+class _ModelState:
+    def __init__(self, aggr: ModelUpdateDXOAggregator):
+        self.aggregator = aggr
+        self.devices = {}
+        self.last_update_time = None
+
+    def accept(self, model_update: ModelUpdate, fl_ctx: FLContext):
+        self.last_update_time = time.time()
+        self.devices.update(model_update.devices)
+        return self.aggregator.accept(model_update.update, fl_ctx)
+
+
+class ModelManager(FLComponent):
```

Comment:
Similarly, ModelManager should be an abstract class that only defines the spec. This allows different implementations.

What you implemented here should be just one impl.

Reply:
abstracted